packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -0,0 +1,182 @@
+  },
+  summaryTitle: {
+    letterSpacing: 1
+  },
+  payToUnlockTitleContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    gap: spacing(2),
+    marginBottom: spacing(2)
+  },
+  buy: {
+    fontSize: typography.fontSize.large,
+    lineHeight: typography.fontSize.large * 1.5
+  },
+  button: {
+    padding: spacing(4),
+    height: spacing(12)
+  }
+}))
+
+export const PremiumTrackPurchaseDrawer = () => {
+  const styles = useStyles()
+  const { neutralLight2, specialLightGreen1 } = useThemeColors()
+  const { data } = useDrawer('PremiumTrackPurchase')
+  const { trackId } = data
+  const track = useSelector((state) => getTrack(state, { id: trackId }))
+  const user = useSelector((state) => getUser(state, { id: track?.owner_id }))
+
+  const handleConfirmPress = useCallback(() => {
+    console.log('buy button pressed')
+  }, [])
+
+  const { premium_conditions: premiumConditions } = track ?? {}
+
+  if (!track || !isPremiumContentUSDCPurchaseGated(premiumConditions) || !user)
+    return null
+
+  const price = formatUSDCWeiToUSDString(premiumConditions.usdc_purchase.price)
+
+  return (
+    <NativeDrawer drawerName={PREMIUM_TRACK_PURCHASE_MODAL_NAME}>
+      <View style={styles.drawer}>
+        <View style={styles.titleContainer}>
+          <IconCart fill={neutralLight2} />
+          <Text style={styles.title}>{messages.title}</Text>
+        </View>
+        <TrackDetailsTile trackId={track.track_id} />
+        <View style={styles.summaryContainer}>
+          <View style={[styles.summaryRow, styles.greyRow]}>
+            <Text
+              weight='bold'
+              textTransform='uppercase'
+              style={styles.summaryTitle}
+            >
+              {messages.summary}
+            </Text>
+          </View>
+          <View style={styles.summaryRow}>
+            <Text>{messages.artistCut}</Text>
+            <Text>{messages.price(price)}</Text>
+          </View>
+          <View style={styles.summaryRow}>
+            <Text>{messages.audiusCut}</Text>
+            <Text>{messages.always0}</Text>
+          </View>
+          <View style={[styles.summaryRow, styles.lastRow, styles.greyRow]}>
+            <Text weight='bold'>{messages.youPay}</Text>
+            <Text weight='bold' color='secondary'>
+              {messages.price(price)}
+            </Text>
+          </View>
+        </View>
+        <View>
+          <View style={styles.payToUnlockTitleContainer}>
+            <Text weight='heavy' textTransform='uppercase' fontSize='small'>
+              {messages.payToUnlock}
+            </Text>
+            <LockedStatusBadge locked={true} />
+          </View>
+          <Text>{messages.disclaimer}</Text>
+        </View>
+        <Button
+          title={messages.buy(price)}
+          onPress={handleConfirmPress}
+          variant={'primary'}
+          styles={{
+            root: styles.button,
+            text: styles.buy
```

Comment:
Are these styles doing something the existing size variants on button can't do? I was hoping we could avoid custom button styling going forward (trying to do this on web by implementing the harmony buttons)